DB/src/main/java/io/deephaven/db/v2/utils/KeyedArrayBackedMutableTable.java

```diff
@@ -204,8 +203,10 @@ protected void processPendingDelete(Table table, IndexChangeRecorder indexChange
         try (final WritableLongChunk<Attributes.KeyIndices> destinations =
                 WritableLongChunk.makeWritableChunk(chunkCapacity)) {
             try (final ChunkSource.GetContext getContext = keySource.makeGetContext(chunkCapacity, sharedContext);
-                    final ChunkBoxer.BoxerKernel boxer = ChunkBoxer.getBoxer(keySource.getChunkType(), chunkCapacity)) {
-                final Chunk<? extends Attributes.Values> keys = keySource.getChunk(getContext, table.getIndex());
+                    final ChunkBoxer.BoxerKernel boxer = ChunkBoxer.getBoxer(keySource.getChunkType(), chunkCapacity);
+                    final Index tableIndex = table.getIndex().clone();) {
```

Comment:
Same question: why do we need to clone an index we aren't mutating or keeping long-term ownership of?

Reply:
The problem is that this index is the table's actual index. There is a call that creates an ordered keys chunk of ranges which is then cached in the Index. But that Chunk is never free'd in the test because it isn't closed until the index itself is closed. So it looks like a chunk-leak to the test.

It seemed like the only way around this was to clone and close -- enabling that orderedkeys cached chunk to be cleaned up.